gadget/gadget.go

```diff
@@ -560,7 +637,7 @@ func validateStructureType(s string, vol *Volume) error {
 	return nil
 }
 
-func validateRole(vs *VolumeStructure, vol *Volume) error {
+func validateRole(vs *VolumeStructure, vol *Volume, constraints *ModelConstraints) error {
```

Comment:
No need to pass the constraints to `validateRole` anymore

Reply:
Removed unnecessary parameter.